Source/WebCore/rendering/MarkedText.cpp

```diff
@@ -122,6 +122,20 @@ Vector<MarkedText> MarkedText::collectForHighlights(const RenderText& renderer,
                 for (auto& rangeData : highlightRegister->map().get(highlightName)->rangesData()) {
                     if (!highlightData.setRenderRange(rangeData))
                         continue;
+                    if (auto* staticRange = dynamicDowncast<StaticRange>(rangeData->range()); staticRange
+                        && (!staticRange->isValid() || !staticRange->computeValidityForPaintingHighlights()))
+                        continue;
+                    // FIXME: Potentially move this check elsewhere, to where we collect this range information.
+                    bool noRenderer = true;
```

Comment:
It's probably more readable to wrap this code in an anonymous lambda like this:
```
auto hasRenderer = [&]() {
    IntersectingNodeRange nodes(makeSimpleRange(rangeData->range()));
    for (auto& iterator : nodes) {
        if (iterator.renderer())
            return true;
    }
    return false;
}();
```

Reply:
Will make this change soon! Thanks, it def looks much better this way.